server/server.go

```diff
@@ -218,17 +218,29 @@ func New(c *Config) (*Server, error) {
 		Base: basePolicyHandler,
 	}
 
+	oauth2Params := []oauth2.Param{
+		oauth2.ForceTLS(forceTLS),
+		oauth2.WithStore(&oauth2.SessionStateStore{
+			Sessions: sessions,
+		}),
+		oauth2.OnLogin(handler.Login(c.Github, basePath, sessions)),
+	}
+	if c.Server.PublicURL != "" {
+		u, err := url.Parse(c.Server.PublicURL)
```

Comment:
This is already parsed as `publicURL` on L73, so I think you can re-use it here. Since the earlier logic also requires that the value is not empty, I think we can unconditionally set the `WithRedirectURL` parameter.